servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpResponseDecoder.java

```diff
@@ -84,10 +97,19 @@ protected HttpResponseMetaData createMessage(final ByteBuf buffer, final int fir
                                                  final int thirdStart, final int thirdLength) {
         return newResponseMetaData(nettyBufferToHttpVersion(buffer, firstStart, firstLength),
                 HttpResponseStatus.of(nettyBufferToStatusCode(buffer, secondStart, secondLength),
-                        thirdLength >= 0 ? buffer.toString(thirdStart, thirdLength, US_ASCII) : ""),
+                        reasonPhrase(buffer, thirdStart, thirdLength)),
                 headersFactory().newHeaders());
     }
 
+    private static String reasonPhrase(final ByteBuf buffer, final int start, final int length) {
+        if (length <= 0) {
+            return EMPTY_STRING;
+        }
+        // Verify reason-phrase = *( HTAB / SP / VCHAR / obs-text )
+        buffer.forEachByte(start, length, ENSURE_NO_CONTROL_CHARS);
```

Comment:
https://tools.ietf.org/html/rfc7230#section-3.1.2
> A client SHOULD ignore the reason-phrase content.

sad to waste cycles validating something that should be ignored 😢

Reply:
We can optimize the "most-likely path" of hitting the known status code if we will make `HttpResponseStatus.valueOf(int)` public. Right know, there is no cheap way to check if we already have a constant for the `status-code` because we always fallback to `new HttpResponseStatus` that is relatively expensive. We can do that in a follow up. Something like:
1. Check if the `status-code` is known.
2. If yes, return the constant. If no, do the validation and create a new `HttpResponseStatus`.